livechat/utils/ws_client.py

```diff
@@ -2,153 +2,96 @@
-                    sslopt={'cert_reqs': ssl.CERT_NONE},
-                    enable_multithread=True)
-            else:
-                self.websocket = websocket.create_connection(
-                    self.url,
-                    self.timeout,
-                    sslopt={'cert_reqs': ssl.CERT_NONE},
-                    enable_multithread=True)
-        except (WebSocketException, WebSocketTimeoutException,
-                WebSocketProtocolException, WebSocketPayloadException,
-                WebSocketConnectionClosedException, WebSocketProxyException,
-                WebSocketBadStatusException,
-                WebSocketAddressException) as exception:
-            self.logger.critical(f'WebSocket Exception: {exception}')
-        else:
-            if keep_alive:
-                self.keep_alive = True
-                threading.Thread(target=self._keep_ws_alive).start()
+            assert self.sock.connected
+            return
+        except (AttributeError, AssertionError):
+            sleep(0.1)
+            return self._wait_till_sock_connected(timeout=timeout - 0.1)
 
-    def close(self) -> None:
-        ''' Close WebSocket connection. '''
-        self.keep_alive = False
-        if self.websocket.connected:
-            try:
-                self.websocket.close()
-            except WebSocketConnectionClosedException as error:
-                self.logger.critical(
-                    f'Exception caught while closing the WebSocket: {error}.')
+    def open(self,
+             origin: dict = None,
+             timeout: float = 3,
+             keep_alive: bool = True):
+        ''' Open websocket connection and keep running forever.
+            Args:
+                origin (dict): Specifies origin while creating websocket connection.
+                timeout (int or float): time [seconds] to wait for server in ping/pong frame.
+                keep_alive(bool): Bool which states if connection should be kept, by default sets to `True`.
+                    origin (dict): Specifies origin while creating websocket connection. '''
+        run_forever_kwargs = {
+            'sslopt': {
+                'cert_reqs': ssl.CERT_NONE
+            },
+            'origin': origin,
+            'ping_timeout': timeout,
+            'ping_interval': 5
+        }
+        if keep_alive:
+            ping_thread = threading.Thread(target=self.run_forever,
+                                           kwargs=run_forever_kwargs)
+            ping_thread.start()
+            self._wait_till_sock_connected()
+            return
+        self.run_forever(**run_forever_kwargs)
 
-    def send(self, request: dict) -> dict:
-        ''' Send request via WebSocket.
-               Args:
-                    request (dict): Dictionary which is being converted to payload.
-               Returns:
-                    dict: Dictionary with response.
+    def send(self, request: dict, opcode=ABNF.OPCODE_TEXT, response_timeout=2):
```

Comment:
Should be: `def send(self, request: dict, opcode=ABNF.OPCODE_TEXT, response_timeout=2) -> dict:`